luxonis_train/attached_modules/metrics/confusion_matrix.py

```diff
@@ -0,0 +1,285 @@
+            num_classes=self.n_classes
+        )
+
+    def prepare(
+        self, inputs: Packet[Tensor], labels: Labels
+    ) -> tuple[dict[str, Tensor], dict[str, Tensor]]:
+        """Prepare data for classification, segmentation, and detection
+        tasks.
+
+        @type inputs: Packet[Tensor]
+        @param inputs: The inputs to the model.
+        @type labels: Labels
+        @param labels: The ground-truth labels.
+        @return: A tuple of two dictionaries: one for predictions and
+            one for targets.
+        """
+        predictions = {}
+        targets = {}
+
+        if self.is_detection:
+            out_bbox = self.get_input_tensors(inputs, TaskType.BOUNDINGBOX)
+            bbox = self.get_label(labels, TaskType.BOUNDINGBOX)
+            bbox = bbox.to(out_bbox[0].device).clone()
+            bbox[..., 2:6] = box_convert(bbox[..., 2:6], "xywh", "xyxy")
+            scale_factors = torch.tensor(
+                [
+                    self.original_in_shape[2],
+                    self.original_in_shape[1],
+                    self.original_in_shape[2],
+                    self.original_in_shape[1],
+                ],
+                device=bbox.device,
+            )
+            bbox[..., 2:6] *= scale_factors
+            predictions["detection"] = out_bbox
+            targets["detection"] = bbox
+
+        if self.is_classification:
+            prediction = self.get_input_tensors(
+                inputs, TaskType.CLASSIFICATION
+            )
+            target = self.get_label(labels, TaskType.CLASSIFICATION).to(
+                prediction[0].device
+            )
+            predictions["classification"] = prediction
+            targets["classification"] = target
+
+        if self.is_segmentation:
+            prediction = self.get_input_tensors(inputs, TaskType.SEGMENTATION)
+            target = self.get_label(labels, TaskType.SEGMENTATION).to(
+                prediction[0].device
+            )
+            predictions["segmentation"] = prediction
+            targets["segmentation"] = target
+
+        return predictions, targets
+
+    def update(
+        self, predictions: dict[str, Tensor], targets: dict[str, Tensor]
+    ) -> None:
+        """Update the confusion matrices for all tasks using prepared
+        data.
+
+        @type predictions: dict[str, Tensor]
+        @param predictions: A dictionary containing predictions for all
+            tasks.
+        @type targets: dict[str, Tensor]
+        @param targets: A dictionary containing targets for all tasks.
+        """
+        if "classification" in predictions and "classification" in targets:
+            preds = predictions["classification"]
+            target = targets["classification"]
+            pred_classes = preds[0].argmax(dim=1)  # [B]
+            target_classes = target.argmax(dim=1)  # [B]
+            self.classification_cm += self.compute_confusion_matrix(
```

Comment:
Would it be better to have self.classification_cm be an actual torchmetric instance and then here in update we call update on it and in compute call compute on it? (same for segmentation_cm)
And since only one of them can be active (since only one task is allowed) can we get away with only one instance for both?